mmocr/models/textrecog/encoders/restransformer.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+
+from mmcv.cnn.bricks.transformer import BaseTransformerLayer
+from mmcv.runner import BaseModule, ModuleList
+
+from mmocr.models.builder import BACKBONES, build_backbone
+from mmocr.models.common.modules import PositionalEncoding
+
+
+@BACKBONES.register_module()
+class ResTransformer(BaseModule):
+    """Implement ResTransformer backbone for text recognition, modified from
+    `<https://github.com/FangShancheng/ABINet>`.
+
+    Args:
+        n_layers (int): Number of attention layers.
+        n_head (int): Number of parallel attention heads.
+        d_model (int): Dimension :math:`D_m` of the input from previous model.
+        d_inner (int): Hidden dimension of feedforward layers.
+        dropout (float): Dropout rate.
+        max_len (int): Maximum output sequence length :math:`T`.
+        res_backbone (dict): Backbone config of ResTransformer. Defaults to
+            dict(type='ResNetABI').
+        init_cfg (dict or list[dict], optional): Initialization configs.
+    """
+
+    def __init__(self,
+                 n_layers=2,
+                 n_head=8,
+                 d_model=512,
+                 d_inner=2048,
+                 dropout=0.1,
+                 max_len=8 * 32,
+                 res_backbone=dict(type='ResNetABI'),
+                 init_cfg=None):
+        super().__init__(init_cfg=init_cfg)
+
+        assert d_model % n_head == 0, 'd_model must be divisible by n_head'
+
+        self.resnet = build_backbone(res_backbone)
+        self.pos_encoder = PositionalEncoding(d_model, n_position=max_len)
+        encoder_layer = BaseTransformerLayer(
+            operation_order=('self_attn', 'norm', 'ffn', 'norm'),
+            attn_cfgs=dict(
+                type='MultiheadAttention',
+                embed_dims=d_model,
+                num_heads=n_head,
+                attn_drop=dropout,
+                dropout_layer=dict(type='Dropout', drop_prob=dropout),
+            ),
+            ffn_cfgs=dict(
+                type='FFN',
+                embed_dims=d_model,
+                feedforward_channels=d_inner,
+                ffn_drop=dropout,
+            ),
+            norm_cfg=dict(type='LN'),
+        )
+        self.transformer = ModuleList(
+            [copy.deepcopy(encoder_layer) for _ in range(n_layers)])
```

Comment:
remove the ResTransformer and only use the ResNetABI for backbone
This transform should be encode of vision branch
```suggestion
    @ENCODE.register_module()
    class ABINetVisionEncode(BaseEncode)
        self.pos_encoder = PositionalEncoding(d_model, n_position=max_len)
        encoder_layer = BaseTransformerLayer(
            operation_order=('self_attn', 'norm', 'ffn', 'norm'),
            attn_cfgs=dict(
                type='MultiheadAttention',
                embed_dims=d_model,
                num_heads=n_head,
                attn_drop=dropout,
                dropout_layer=dict(type='Dropout', drop_prob=dropout),
            ),
            ffn_cfgs=dict(
                type='FFN',
                embed_dims=d_model,
                feedforward_channels=d_inner,
                ffn_drop=dropout,
            ),
            norm_cfg=dict(type='LN'),
        )
        self.transformer = ModuleList(
            [copy.deepcopy(encoder_layer) for _ in range(n_layers)])
```